persistent/Database/Persist/Quasi/Internal.hs

```diff
@@ -226,14 +237,50 @@ lowestIndent = minimum . fmap lineIndent
 
 -- | Divide lines into blocks and make entity definitions.
 parseLines :: PersistSettings -> NonEmpty Line -> [EntityDef]
-parseLines ps =
-    fixForeignKeysAll . map mk . associateLines
+parseLines ps = do
+    fixForeignKeysAll . fmap (mkEntityDef ps . toParsedEntityDef) . associateLines
+
+data ParsedEntityDef = ParsedEntityDef
+    { parsedEntityDefComments :: [Text]
+    , parsedEntityDefEntityName :: EntityNameHS
+    , parsedEntityDefIsSum :: Bool
+    , parsedEntityDefEntityAttributes :: [Attr]
+    , parsedEntityDefFieldAttributes :: [[Token]]
+    , parsedEntityDefExtras :: M.Map Text [ExtraLine]
+    }
```

Comment:
Oh this is *fantastic*

In #1256 I'm probably going to be only forwarding this type along, instead of the `EntityDef`. Maybe the `UnboundEntityDef`.